### PR TITLE
Manage AI crawlers

### PR DESCRIPTION
**Pull request by jitu5**
## Description

Resolves https://github.com/kedro-org/kedro-website/issues/198

This pull request introduces several changes to manage web crawlers more effectively by updating the `robots.txt` file and handling user-agent requests. The most important changes include the addition of allowed and disallowed agents, generating `robots.txt` dynamically, and implementing edge functions to handle user-agent requests.

## Development notes

* [`agents.json`](diffhunk://#diff-15e11c2fca509b2f191942245c97d89c01d3b21735e8c6b7a85ab5044b3f8bcdR1-R37): Added lists of allowed and disallowed user agents for web crawlers.
* [`generateRobotsTxt.js`](diffhunk://#diff-f2412c2942de4a6865f1afc3a136824a3d15e4d09a826750b50f205edc1998c5R1-R26): Created a script to dynamically generate `robots.txt` based on the agents listed in `agents.json` when `npm run build` is executed. 
* [`netlify/edge-functions/bots.ts`](diffhunk://#diff-9a0474688c2a5261149394ff63789f7b34816ae0230cbec4d75922ea7a1fb8e5R1-R31): Implemented an edge function to handle requests based on user-agent headers, blocking disallowed agents and allowing others.

* [`package.json`](diffhunk://#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L5-R5): Updated the build script to include the generation of `robots.txt` before building the project.
* [`package.json`](diffhunk://#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R17): Added `@netlify/edge-functions` as a dependency to support the new edge function.

**Note:**
As of now I am not fully sure which to allow and disallowed but I created below list from [ai.robots.txt](https://github.com/ai-robots-txt/ai.robots.txt) and [darkvisitors.com](https://darkvisitors.com)

```
{
    "allowed": [
      "Googlebot",
      "Bingbot",
      "Applebot",
      "ChatGPT",
      "ChatGPT-User"
    ],
    "disallowed": [
      "Claude-Web",
      "ClaudeBot",
      "AI2Bot",
      "anthropic-ai",
      "AwarioRssBot",
      "AwarioSmartBot",
      "Bytespider",
      "CCBot",
      "diffbot",
      "PerplexityBot",
      "Omgili",
      "ImagesiftBot",
      "Youbot",
      "Applebot-Extended",
      "Facebookbot",
      "FacebookBot",
      "Meta-ExternalAgent",
      "Meta-ExternalFetcher",
      "Kangaroo Bot",
      "Timpibot",
      "Webzio-Extended",
      "DuckAssistBot",
      "OAI-SearchBot",
      "Amazonbot",
      "Google-Extended"
    ]
  }
```

### CODE REVIEW

File: agents.json
Comment (tynandebold) [inline agents.json:1]: Personal bias here: I use Claude/Anthropic and Perplexity. I'd suggest those ones can be allowed.

Comment (rashidakanchwala) [reply agents.json:1]: Yeah, we should try adding them! I am sure the above companies must be doing guarding against crazy crawling, and we can monitor

**Review by rashidakanchwala — APPROVED** [review]
Nice, I guess once we merge we can test it on the Kedro website if the edge function is running  :) .. agree to add the allowed bots Tynan suggested. thanks a lot!!

**Review by tynandebold — APPROVED** [review]
Nice!

File: agents.json [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      "Applebot",
+      "ChatGPT",
+      "ChatGPT-User",
+      "Claude-Web",
+      "ClaudeBot",
+      "PerplexityBot"
+    ],
+    "disallowed": [
+      "AI2Bot",
+      "anthropic-ai",
```
Comment (astrojuanlu) [inline agents.json:14]: Maybe allow `anthropic-ai`?

**Review by astrojuanlu — APPROVED** [review]
💯

**Pull request merged**